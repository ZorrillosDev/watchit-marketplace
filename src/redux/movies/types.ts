export interface MoviesArgs {
  limit?: number
}
