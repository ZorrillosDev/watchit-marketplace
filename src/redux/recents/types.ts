export interface RecentMoviesArgs {
    limit?: number
}