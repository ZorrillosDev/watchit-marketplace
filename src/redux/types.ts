import { ThunkAction as AsyncThunkAction, ThunkDispatch } from 'redux-thunk'
import { AnyAction } from 'redux'
import { RootState } from '@state/store'

export type ThunkAction<R> = AsyncThunkAction<R, RootState, any, AnyAction>
export type ThunkDispatcher = ThunkDispatch<RootState, any, AnyAction>

export const GenresList = [
  'All',
  'Action',
  'Adventure',
  'Animation',
  'Biography',
  'Comedy',
  'Crime',
  'Documentary',
  'Drama',
  'Family',
  'Fantasy',
  'Film-Noir',
  'History',
  'Horror',
  'Music',
  'Musical',
  'Mystery',
  'Romance',
  'Sci-Fi',
  'Sport',
  'Thriller',
  'War',
  'Western',
  'News',
  'Reality-TV',
  'Talk-Show',
  'Game-Show'
] as const

export type Genres = typeof GenresList[number]

export interface MovieProperties {
  date_uploaded_unix: number
  genres: Genres
  group_name: string
  imdb_code: string
  language: string
  mpa_rating: string
  rating: number
  runtime: number
  synopsis: string
  title: string
  trailer_code: string
  year: number
}

export interface Movie {
  creator: string
  description: string
  image: string
  name: string
  properties: MovieProperties
}
