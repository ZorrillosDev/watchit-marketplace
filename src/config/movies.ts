export const FAKE_MOVIES = [
  {
    creator: {
      address: '0x0010',
      movies: 4
    },
    owner: {
      address: '0x0022',
      movies: 9
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/11/help-50817-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  },
  {
    creator: {
      address: '0xEe99CeFF640d37Edd9cac8c7cfF4Ed4cD609f435',
      movies: 10
    },
    owner: {
      address: '0x0054',
      movies: 7
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/08/safer-at-home-46847-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  },
  {
    creator: {
      address: '0x005',
      movies: 5
    },
    owner: {
      address: '0x006',
      movies: 3
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/11/cagefighter-worlds-collide-50877-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  },
  {
    creator: {
      address: '0x007',
      movies: 2
    },
    owner: {
      address: '0x008',
      movies: 12
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/10/este-nino-es-un-trasto-50575-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  }
]
