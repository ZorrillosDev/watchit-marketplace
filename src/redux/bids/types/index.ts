export * from './bids';
export * from './bids.actions';
export * from './bids.state';
