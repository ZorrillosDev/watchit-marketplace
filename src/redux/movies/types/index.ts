export * from './movies'
export * from './movies.actions'
export * from './movies.state'