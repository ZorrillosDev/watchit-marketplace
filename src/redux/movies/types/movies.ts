export const GenresList = [
  'All',
  'Action',
  'Adventure',
  'Animation',
  'Biography',
  'Comedy',
  'Crime',
  'Documentary',
  'Drama',
  'Family',
  'Fantasy',
  'Film-Noir',
  'History',
  'Horror',
  'Music',
  'Musical',
  'Mystery',
  'Romance',
  'Sci-Fi',
  'Sport',
  'Thriller',
  'War',
  'Western',
  'News',
  'Reality-TV',
  'Talk-Show',
  'Game-Show'
] as const

export type Genres = typeof GenresList[number]

export interface Images {
  small: string
  medium: string
  large: string
}

export interface MovieBid {
  id: string
  account: string
  created_at: string
  bid: number
}

export interface Movie {
  creator: string
  description: string
  path: string
  title: string
  date_uploaded_unix: number
  price: number
  group_name: string
  imdb_code: string
  language: string
  mpa_rating: string
  rating: number
  runtime: number
  synopsis: string
  trailer_code: string
  year: number
  genres: Genres
  posters: Images
}
