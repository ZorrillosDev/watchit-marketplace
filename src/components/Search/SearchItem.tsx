// REACT IMPORTS
import React, { FC } from 'react';

// MUI IMPORTS
import { Grid, GridProps, styled, Typography } from '@mui/material';

// PROJECT IMPORTS
import { Color } from '@src/utils';
import { Movie } from '@state/movies/types';
import TruncatedTypography from '@components/TruncatedTypography';
import Poster from '@components/Poster';
import { IconStar } from '@tabler/icons';

// ===========================|| NOTIFICATIONS - ITEM ||=========================== //

const SearchItem: FC<Movie> = ({ ...movie }): JSX.Element => {
  return (
    <SearchItemWrapper item xs={12} display='flex' sx={{ p: '0' }}>
      <Poster image={movie.posters.small} name={movie.title} />
      <Grid container sx={{ p: 1 }}>
        <Grid item xs={9}>
          <Grid container>
            <Grid item xs={12}>
              <Typography variant='h4' color='primary'>
                {movie.title}
              </Typography>
            </Grid>
            <Grid item xs={12}>
              <TruncatedTypography lines={4} sx={{ m: 0 }}>
                {movie.synopsis}
              </TruncatedTypography>
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={3}>
          <Grid container>
            <Grid item xs={12}>
              <Typography variant='h5' color='secondary' textAlign='right' fontWeight='bold'>
                {movie.year}
              </Typography>
            </Grid>
            <Grid item xs={12}>
              <Typography variant='h5' color='secondary' textAlign='right'>
                <IconStar />
                {movie.rating}
              </Typography>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </SearchItemWrapper>
  );
};

const SearchItemWrapper = styled(Grid)<GridProps>(({ theme }) => ({
  cursor: 'pointer',
  padding: '16px',
  '& > .MuiCard-root': {
    width: '6rem',
  },
  '& svg': {
    display: 'inline',
    width: '1rem',
    height: '1rem',
    marginRight: '0.25rem',
    transform: 'translateY(3px)',
  },
  '&:hover': {
    background: Color.addAlpha(theme.palette.primary.light, 0.3),
  },
  '&:last-of-type': {
    borderBottomRightRadius: theme.shape.borderRadius,
    borderBottomLeftRadius: theme.shape.borderRadius,
  },
}));

export default SearchItem;
