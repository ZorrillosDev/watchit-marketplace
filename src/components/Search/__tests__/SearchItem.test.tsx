import SearchItem from '@components/Search/SearchItem'
import { shallow } from 'enzyme'
import React from 'react'
import { Movie } from '@state/movies/types'

/* eslint-disable  @typescript-eslint/consistent-type-assertions */

describe('<SearchItem />', () => {
  it('should render', () => {
    const component = shallow(
      <SearchItem {...{ posters: { small: 'test' } } as Movie} />
    )
    expect(component).toMatchSnapshot()
  })
})
