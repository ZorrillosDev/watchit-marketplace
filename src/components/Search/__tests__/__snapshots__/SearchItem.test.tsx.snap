// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchItem /> should render 1`] = `ShallowWrapper {}`;
